libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/ImmediateRouteRefreshController.kt

```diff
@@ -0,0 +1,38 @@
+package com.mapbox.navigation.core.routerefresh
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+internal class ImmediateRouteRefreshController(
+    private val routeRefresherExecutor: RouteRefresherExecutor,
+    private val plannedRefreshController: Pausable,
+    private val stateHolder: RouteRefreshStateHolder,
+    private val listener: RouteRefresherListener
+) {
+
+    private val callback = object : RouteRefresherProgressCallback {
+
+        override fun onStarted() {
+            stateHolder.onStarted()
+        }
+
+        override fun onResult(routeRefresherResult: RouteRefresherResult) {
+            if (routeRefresherResult.success) {
+                stateHolder.onSuccess()
+            } else {
+                stateHolder.onFailure(null)
+                plannedRefreshController.resume()
```

Comment:
Because success result means new refreshed routes -> we don't have to resume refreshing the old ones. We'll get a notification with reason = REFRESH and relaunch planned refresh.